compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureGenerator.java

```diff
@@ -587,17 +587,23 @@ BLangLambdaFunction createLambdaFunction(BLangFunction function) {
 
     public BInvokableSymbol createSimpleVariable(BLangFunction function, BLangLambdaFunction lambdaFunction,
                                                  boolean isAnnotationClosure) {
+        Location pos;
+        if (isAnnotationClosure) {
+            pos = env.enclPkg.pos;
```

Comment:
Don't these update the pos of non-module level variables also?

Reply:
No, The `isAnnotationClosure` flag is used to identify closures at the module level.